cib/remote.c

```diff
@@ -324,7 +324,9 @@ cib_remote_listen(gpointer data)
 
     crm_client_init();
     new_client = calloc(1, sizeof(crm_client_t));
+    CRM_ASSERT(new_client != NULL);
     new_client->remote = calloc(1, sizeof(crm_remote_t));
```

Comment:
After a bit of research, I found that in the C standard, dereferencing a NULL pointer results in "undefined behavior".

From http://en.wikipedia.org/wiki/Null_pointer: "Dereferencing the NULL pointer typically results in ... a segmentation fault ... There are, however, certainly circumstances where this is not the case. ... It is also possible for the compiler to optimize away the `NULL` pointer dereference, avoiding a segmentation fault but causing other undesired behavior. ... an attempt to dereference a null pointer usually (but not always) causes a run-time error or immediate program crash."

From http://blog.llvm.org/2011/05/what-every-c-programmer-should-know.html: "contrary to popular belief, dereferencing a null pointer in C is undefined. It is not defined to trap ... In C-based languages, NULL being undefined enables a large number of simple scalar optimizations that are exposed as a result of macro expansion and inlining."